master/master.go

```diff
@@ -169,20 +172,49 @@ func lineNums(file string) int {
 	return num
 }
 
+func orDone(finish, crashChan <-chan string, numberOfTasks int) <-chan string {
+	count := 0
+	valStream := make(chan string)
+	go func() {
+		defer close(valStream)
+		for {
+			select {
+			case <-finish:
+				count += 1
+				if count == numberOfTasks {
+					return
+				}
+			case crashUUID, ok := <-crashChan:
+				if !ok {
+					if count != numberOfTasks {
+						log.Panic("faild to distribute all Tasks")
+						return
+					}
+				}
+
+				valStream <- crashUUID
+
+			}
+		}
+	}()
+	return valStream
+}
+
```

Comment:
This looks weird to me.
As far as I know, The go routine function should end earlier than the orDone function.

Reply:
Where we read from orDone func will block until we write on valSream channel or close it(this is because of channel attributes)so orDone func and it's groutine won't close until we return

Reply:
Wow, I got it. I think this design is excellent for readability!